iceoryx_hoofs/include/iceoryx_hoofs/error_handling/error_handling.hpp

```diff
@@ -17,289 +17,26 @@
-    /// Log error entry + Assert + terminate
-    FATAL,
-    /// warning log entry + Assert
-    SEVERE,
-    /// warning log entry
-    MODERATE
+    NO_ERROR = HOOFS_MODULE_IDENTIFIER << ERROR_ENUM_OFFSET_IN_BITS,
+    HOOFS_ERRORS(CREATE_ICEORYX_ERROR_ENUM)
 };
 
-/// @brief Howto use the error handler correctly
-///     1.) Use the macro ICEORYX_ERRORS(error) to create the enum for your component and
-///             add new errors like:
-///             error(MODULE_NAME__MY_FUNKY_ERROR)
-///         Attention: Create an error after the following convention:
-///             MODULE_NAME__A_CLEAR_BUT_SHORT_ERROR_DESCRIPTION
-///         And a long name is alright!
-///
-///     2.) Specialize the following methods for your NewEnumErrorType:
-///         - const char* toString(const NewEnumErrorType error)
-///
-///     3.) Call errorHandler(Error::kMODULE_NAME__MY_FUNKY_ERROR);
-///             Please pay attention to the "k" prefix
-///         The defaults for errorCallback and ErrorLevel can also be overwritten:
-///             errorHandler(
-///                 Error::kMODULE_NAME__MY_FUNKY_ERROR,
-///                 []{ std::cout << "MyCustomCallback" << std::endl; },
-///                 ErrorLevel::MODERATE
-///             );
-///
-/// @code
-/// class PrettyClass {
-///     float division(float a, float b) {
-///         if ( b == 0.0f ) {
-///             errorHandler(Error::kPRETTY_CLASS__DIVISION_BY_ZERO);
-///         }
-///     }
-/// };
-/// @endcode
-///
-/// @code
-/// bool called = false;
-/// auto temporaryErrorHandler = ErrorHandler::setTemporaryErrorHandler(
-///     [&](const Error e, std::function<void()>, const ErrorLevel) {
-///         called = true;
-///     });
-///
-/// errorHandler(Error::kTEST__ASSERT_CALLED);
-/// ASSERT_TRUE(called);
-/// @endcode
-/// @tparam[in] Error type which is used to report the error (typically an enum)
-template <typename Error>
-void errorHandler(const Error error,
-                  const std::function<void()>& errorCallBack = std::function<void()>(),
-                  const ErrorLevel level = ErrorLevel::FATAL) noexcept;
-
-using HandlerFunction = std::function<void(const uint32_t, const char*, const ErrorLevel)>;
-
-/// @brief This handler is needed for unit testing, special debugging cases and
-///         other corner cases where we'd like to explicitly suppress the
-///         error handling.
-class ErrorHandler
-{
-    template <typename Error>
-    friend void
-    errorHandler(const Error error, const std::function<void()>& errorCallBack, const ErrorLevel level) noexcept;
-
-  protected:
-    static void reactOnErrorLevel(const ErrorLevel level, const char* errorText) noexcept;
-
-    static void
-    defaultHandler(const uint32_t error, const char* errorName, const ErrorLevel level = ErrorLevel::FATAL) noexcept;
-
-    static iox::HandlerFunction handler;
-};
-
-const char* toString(const Error error) noexcept;
+const char* toString(const HoofsError error) noexcept;
```

Comment:
Can this be inlined?

Also, if you rename this to `asStringLiteral` if will automatically have a LogStream operator implemented with #1066

Reply:
Renamed the functions to `asStringLiteral`. However, didn't inline the implementation as this is no templated code and shall therefore reside in the `.cpp`